src/main/scala/firrtl/transforms/PropagatePresetAnnotations.scala

```diff
@@ -71,7 +71,7 @@ class PropagatePresetAnnotations extends Transform with DependencyAPIMigration {
     * @param presetAnnos all the annotations
     * @return updated annotations
     */
-  private def propagate(cs: CircuitState, presetAnnos: Seq[PresetAnnotation]): AnnotationSeq = {
+  private def propagate(cs: CircuitState, presetAnnos: Seq[PresetAnnotation], otherAnnos: Seq[Annotation]): AnnotationSeq = {
```

Comment:
not a good idea to change API here.

Reply:
This is a private method, so no externally visible API change. The problem was the way that the annotations were filtered was broken. I think the `partition` pattern is the right way to do it. However, I could be persuaded to filter out the `otherAnnos` inside the `propagate` method.